modules/operator/main.tf

```diff
@@ -0,0 +1,188 @@
+locals {
+  name_prefix = "${var.namespace}-${var.environment}"
+}
+
+resource "kubernetes_namespace" "materialize" {
+  metadata {
+    name = var.operator_namespace
+  }
+}
+
+resource "kubernetes_namespace" "instance_namespaces" {
+  for_each = toset(compact([for instance in var.instances : instance.namespace if instance.namespace != null]))
+
+  metadata {
+    name = each.key
+  }
+}
+
+resource "helm_release" "materialize_operator" {
+  name      = "${local.name_prefix}-operator"
+  namespace = kubernetes_namespace.materialize.metadata[0].name
+  # TODO: Publish the chart to a public repository, currently using a forked version of the chart
+  repository = "https://raw.githubusercontent.com/bobbyiliev/materialize/refs/heads/helm-chart-package/misc/helm-charts"
```

Comment:
The main next step here is to decide on how we would like to publish the Helm Chart.

Currently, the PR uses a temporary solution with a forked version of the Materialize Helm chart.

We need to decide on an appropriate public repository for publishing the Helm chart, this could totally live in the main repository, we would just need an extra flow to package and generate the index.yaml:

```sh
helm package operator
helm repo index .
```

Reply:
I vote for main repository because that's easiest for having compatibility and running e2e tests immediately when something in the terraform plugin changes.

Reply:
Yes, this should be doable, in this case, we could use GitHub pages as suggested on the Helm website: https://helm.sh/docs/topics/chart_repository/#github-pages-example